src/docfx.website.themes/default/styles/docfx.js

```diff
@@ -523,6 +466,141 @@ $(function () {
     }
   })();
 
+  // Show footer
+  (function () {
+    initFooter();
+    $(window).on("scroll", showFooter);
+
+    function initFooter() {
+      if (needFooter()) {
+        shiftUpBottomCss();
+        $("footer").show();
+      } else {
+        resetBottomCss();
+        $("footer").hide();
+      }
+    }
+
+    function showFooter() {
+      if (needFooter()) {
+        shiftUpBottomCss();
+        $("footer").fadeIn();
+      } else {
+        resetBottomCss();
+        $("footer").fadeOut();
+      }
+    }
+
+    function needFooter() {
+      var scrollHeight = $(document).height();
+      var scrollPosition = $(window).height() + $(window).scrollTop();
+      return (scrollHeight - scrollPosition) < 1;
+    }
+
+    function resetBottomCss() {
+      $(".sidetoc").removeClass("shiftup");
+      $(".sideaffix").removeClass("shiftup");
+    }
+
+    function shiftUpBottomCss() {
+      $(".sidetoc").addClass("shiftup");
+      $(".sideaffix").addClass("shiftup");
+    }
+  })();
+
+  // For LOGO SVG
+  // Replace SVG with inline SVG
+  // http://stackoverflow.com/questions/11978995/how-to-change-color-of-svg-image-using-css-jquery-svg-image-replacement
+  jQuery('img.svg').each(function () {
+    var $img = jQuery(this);
+    var imgID = $img.attr('id');
+    var imgClass = $img.attr('class');
+    var imgURL = $img.attr('src');
+
+    jQuery.get(imgURL, function (data) {
+      // Get the SVG tag, ignore the rest
+      var $svg = jQuery(data).find('svg');
+
+      // Add replaced image's ID to the new SVG
+      if (typeof imgID !== 'undefined') {
+        $svg = $svg.attr('id', imgID);
+      }
+      // Add replaced image's classes to the new SVG
+      if (typeof imgClass !== 'undefined') {
+        $svg = $svg.attr('class', imgClass + ' replaced-svg');
+      }
+
+      // Remove any invalid XML tags as per http://validator.w3.org
+      $svg = $svg.removeAttr('xmlns:a');
+
+      // Replace image with new SVG
+      $img.replaceWith($svg);
+
+    }, 'xml');
+  });
+
+  previewPageRefresh();
+})
+
+// This function is extracted for docfx preview
+function previewPageRefresh() {
```

Comment:
offline discuss, this previewPageRefresh function is for grouping existing function.

---
In reply to: [104092081](https://github.com/dotnet/docfx/pull/1379#discussion_r104092081) [](ancestors = 104092081)